### PR TITLE
OkEx heartbeat monitoring

### PR DESCRIPTION
**Pull request by chrism2671**
`OkEx.Stream.Connection` sends pings but doesn't monitor if matching pongs are received.

This can sometimes lead to stale data, and it takes ~20 minutes before websockex quits. You can see this on the chart in the flat regions. At the end of flat period, you get a `Tai.StreamDisconnect`, and everything starts moving again.

Flat regions of about 20 mins show stale prices

This change means that if a pong hasn't been received in the last 30 seconds, terminate the socket.

Notes:

- My own version of Tai has diverged from master, so I've done my best to port over my changes. I wasn't able to test it with master, but it works just fine with my version.
- This is my first PR ever, so please let me know if I've done anything wrong or could do better.

Thanks!

### CODE REVIEW

**Comment by rupurt** [comment]
@chrism2671 oh also, do you mind submitting future PR's from a feature branch? It looks like the CI status badge picks up the status from forks when they're submitted on the master branch

**Comment by rupurt** [comment]
@chrism2671 updated the Github action so that your tests should no pass. Can you rebase with `master`. Also can you squash your 3 commits into 1?

**Comment by chrism2671** [comment]
Thanks! It's exciting to be able to contribute something!

I've made the changes, hopefully it's a bit better now :)

File: apps/tai/lib/tai/venue_adapters/okex/stream/connection.ex [hunk 1]
```diff
@@ -183,4 +195,13 @@ defmodule Tai.VenueAdapters.OkEx.Stream.Connection do
...
     |> GenServer.cast({msg, Timex.now()})
   end
+
+  @max_pong_age_s 30
+  defp pong_check(state) do
+    (time_now() - state.last_pong) < @max_pong_age_s
+  end
+
+  defp time_now() do
+    :erlang.system_sime(:seconds)
```
Comment (rupurt) [inline apps/tai/lib/tai/venue_adapters/okex/stream/connection.ex]: Typo 😄 `system_sime` -> `system_time`

Comment (chrism2671) [reply apps/tai/lib/tai/venue_adapters/okex/stream/connection.ex]: Oops!! :) fixed! :)

**Comment by rupurt** [comment]
@chrism2671 💥. I think I posted this in your previous issue, but if you every publish something that's open source that uses `tai`, I'd love to add it to the [`built with tai`](https://github.com/fremantle-capital/tai/blob/master/docs/BUILT_WITH_TAI.md) section.

**Pull request merged**